docs/docs/api/promise.coroutine.md

```diff
@@ -13,7 +13,9 @@ title: Promise.coroutine
 Promise.coroutine(GeneratorFunction(...arguments) generatorFunction, Object options) -> function
 ```
 
-Returns a function that can use `yield` to yield promises. Control is returned back to the generator when the yielded promise settles. This can lead to less verbose code when doing lots of sequential async calls with minimal processing in between. Requires node.js 0.12+, io.js 1.0+ or Google Chrome 40+.
+In a typical use case, the parameter `generatorFunction` should be a `GeneratorFunction` producing a generator that yields one or more `Promise`s. (Also see the Tips section below.)
+
+This method returns a function that returns a `Promise`(TODO: Elaborate this). When the returned function (e.g. the `ping` function in the example below) is called, it will start the generator (by calling its `next()`) and return immediately. Control is returned back to the generator when each yielded promise settles. This can lead to less verbose code when doing lots of sequential async calls with minimal processing in between. Requires node.js 0.12+, io.js 1.0+ or Google Chrome 40+.
```

Comment:
- What will the promise be like when multiple values or promises are yielded?
- We can use `await` now, so that kind of use case will be very rare. Also, the example didn't make use of the returned promise.